src/kit.ts

```diff
@@ -769,6 +848,13 @@ export async function effectiveKitEnvironment(kit: Kit, opts?: expand.ExpansionO
           util.map(util.chain(host_env, kit_env, vs_vars), ([k, v]): [string, string] => [k.toLocaleUpperCase(), v]));
     }
   }
+  if (kit.environmentVariablesShellScript) {
+    const shell_vars = await getShellScriptEnvironment(kit);
+    if (shell_vars) {
+      // replace hostvars with variables set by shell script
```

Comment:
This means you can't add your own custom variables script if you are using the Visual Studio compiler.  Is that intentional?

Reply:
No, that was not intentional. I moved the if (environmentSetupScript) statement (which sets the host_env variable) before the if (visualStudio) statement.